### DISCUSSION TITLE
Raspberry pi pico rp2040 as Tinyusb mass storage?

### DISCUSSION THREAD

**Discussion by Therealjosephchrzempiec** [adafruit/Adafruit_TinyUSB_Arduino#309] [General]
Hello, I wonder if I can use the rp2040 pico as a mass storage device using the library and example sketch [https://github.com/adafruit/Adafruit_TinyUSB_Arduino/tree/master/examples/MassStorage/msc_external_flash](https://github.com/adafruit/Adafruit_TinyUSB_Arduino/tree/master/examples/MassStorage/msc_external_flash)?

**Comment by hathach** [comment]
yes

**Reply by Therealjosephchrzempiec** [reply]
Thank you. I didn't see that until I scrolled down and saw that. I did manage to try that and it will giving me some error missing liveries I downloaded and it did compile. I did manage to get it working. 

One thing I was wondering somewhere it states I can use a external flash. How do I enable that to use a external winbond of my own?

**Reply by smalltoe** [reply]
Hello,
Did you get that example working with the external flash? I came as far as this here: https://github.com/smalltoe/spi_flash_test
